src/update-process-env.js

```diff
@@ -0,0 +1,49 @@
+/** @babel */
+
+import {spawnSync} from 'child_process'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set(['NODE_ENV', 'NODE_PATH'])
+
+export default function updateProcessEnv (launchEnv) {
+  let envToAssign
+  if (launchEnv.PWD) {
+    envToAssign = launchEnv
+  } else {
+    if (process.platform === 'darwin') {
+      envToAssign = getEnvFromShell()
+    }
+  }
+
+  if (envToAssign) {
+    for (let key in process.env) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        delete process.env[key]
+      }
+    }
+
+    for (let key in envToAssign) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        process.env[key] = envToAssign[key]
+      }
+    }
+  }
+}
+
+function getEnvFromShell () {
```

Comment:
@maxbrunsfeld would you guys be interested in replacing this homegrown function with [`consistent-env`](https://www.npmjs.com/package/consistent-env). It supports both sync/async (no startup delay) and a wide variety of shells and is well tested and robust